services/provider/client/auth_interceptor.go

```diff
@@ -0,0 +1,41 @@
+package client
+
+import (
+	"context"
+
+	"github.com/red-hat-storage/ocs-operator/services/provider/common"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"k8s.io/klog"
+)
+
+type AuthInterceptor struct {
+	accessToken string
+}
+
+func NewAuthInterceptor(accessToken string) *AuthInterceptor {
+	return &AuthInterceptor{accessToken}
+}
+
+func (interceptor *AuthInterceptor) Unary() grpc.UnaryClientInterceptor {
+	return func(
+		ctx context.Context,
+		method string,
+		req, reply interface{},
+		cc *grpc.ClientConn,
+		invoker grpc.UnaryInvoker,
+		opts ...grpc.CallOption,
+	) error {
+		klog.Infof("--> unary interceptor: %q", method)
```

Comment:
Do we need the `-->` in the error message?

Reply:
removed.